authorization/src/main/kotlin/com/ritense/authorization/deployment/PermissionDeployer.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ *  Licensed under EUPL, Version 1.2 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.ritense.authorization.deployment
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.ritense.authorization.PermissionRepository
+import com.ritense.valtimo.contract.importchangelog.ChangesetDeployer
+import com.ritense.valtimo.contract.importchangelog.ChangesetDetails
+
+class PermissionDeployer(
+    private val objectMapper: ObjectMapper,
+    private val permissionRepository: PermissionRepository,
+) : ChangesetDeployer {
+
+    override fun getType() = "permission"
+
+    override fun getChangesetDetails(filename: String, content: String): ChangesetDetails {
+        val permissionChangeset = objectMapper.readValue<PermissionChangeset>(content)
+        return ChangesetDetails(
+            permissionChangeset.changesetId,
+            permissionChangeset.permissions
+        )
+    }
+
+    override fun deploy(content: String) {
+        val permissionChangeset = objectMapper.readValue<PermissionChangeset>(content)
+        permissionChangeset.permissions
+            .map { it.toPermission() }
+            .forEach { permission ->
+                permissionRepository.deleteAllByResourceTypeAndActionAndRoleKey(
```

Comment:
The combination of `resourceType`, `action` and `roleKey` is not unique. I don't think we should be deleting anything from the database at this step.

Reply:
A delete statement is necessary to be able to make changes in the permissions